### PR TITLE
Add more content to the docs

### PR DESCRIPTION
**Pull request by mavam**
This PR adds more content to explain how Tenzir works and expands on the reference.

### CODE REVIEW

**Comment by mavam** [comment]
@jachris this is at a point where I'd like to merge it and do another follow-up PR.

File: web/docs/tql2/functions/replace.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+### `pattern: string`
+
+The pattern to replace in `x`.
+
+### `replacement: string`
+
+The replacement value for `pattern`.
+
+### `max = string` (optional)
```
Comment (raxyte) [inline web/docs/tql2/functions/replace.md:27]: `(optional)` is inside the back-ticks on the operator pages, should be consistent here.

Comment (mavam) [reply web/docs/tql2/functions/replace.md:27]: Curious: we are putting a question mark after the type in the diagnostic. In the docs, we use `[key=string]` in the synopsis. We haven't found a good solution for this yet.

Comment (jachris) [reply web/docs/tql2/functions/replace.md:27]: Yes, the question mark was the original design, and we tried to improve it with the docs. Short-term, the diagnostics should be aligned with the docs. Long-term, we can see whether there is an even better solution.

Comment (mavam) [reply web/docs/tql2/functions/replace.md:27]: So how should we document this then?

```markdown
### [max=string]
```

or

```markdown
### `max = string (optional)`
```

or

```markdown
### `max = string` (optional)
```

I'm also torn about the extra spacing we have, e.g., `key: string` and `key = string` in the docs, versus `key:string` and `key=string` in the synopsis and diagnostics.

I think we should agree on one canonical form.

Comment (jachris) [reply web/docs/tql2/functions/replace.md:27]: What we previously committed to is the format as you can see it in the operator docs, including its spacing:
```md
### `max = string (optional)`
```
We can in the future agree to change it, but that's what it is right now.

Comment (mavam) [reply web/docs/tql2/functions/replace.md]: This makes the least sense semantically, as the "optional" suffix is not code. But for consistency I switched to this style.

**Review by tobim — APPROVED** [review]

**Pull request merged**